trie/secure_trie.go

```diff
@@ -40,9 +40,10 @@ var secureKeyPrefix = []byte("secure-key-")
 type SecureTrie struct {
 	*Trie
 
-	hash       hash.Hash
-	secKeyBuf  []byte
-	hashKeyBuf []byte
+	hash        hash.Hash
+	hashKeyBuf  []byte
+	secKeyBuf   []byte
+	secKeyCache map[string][]byte
```

Comment:
No, Go only allows immutable types as keys. A byte slice is not immutable. Converting it into a string actually creates a copy of the content in the background and makes it immutable.